src/handlers/int-test-support/helpers/athenaHelper.ts

```diff
@@ -100,17 +100,31 @@ export const getQueryResults = async <TResponse>(
 };
 
 export const waitAndGetQueryResults = async <TResponse>(
-  queryId: string
+  queryId: string,
+  query: DatabaseQuery
 ): Promise<TResponse[]> => {
-  await poll(
-    async () => await getQueryExecutionStatus(queryId),
-    (result) => result?.state?.match("SUCCEEDED") !== null,
-    {
-      timeout: 65000,
-      interval: 5000,
-      notCompleteErrorMessage: "Query did not succeed within the given timeout",
+  const maxRetries = 3;
+  let retryCount = 0;
+
+  const pollQueryExecutionStatus = async (): Promise<boolean> => {
+    const queryStatus = await getQueryExecutionStatus(queryId);
+    if (
+      queryStatus?.state === "FAILED" &&
+      queryStatus?.stateChangeReason?.includes("NoSuchKey") &&
```

Comment:
This failure was intermittent and was not able to test End2End so unit tested the function waitAndGetQueryResults as below 
```
import { getQueryExecutionStatus, startQueryExecutionCommand, getQueryResults } from './athenaHelper';
import { waitAndGetQueryResults } from "./test";

jest.mock("./athenaHelper", () => {
    const original = jest.requireActual("./athenaHelper");
    return {
        ...original,
        getQueryExecutionStatus: jest.fn(),
        startQueryExecutionCommand: jest.fn(),
        getQueryResults: jest.fn()

    };
});
const mockedGetQueryExecutionStatus = getQueryExecutionStatus as jest.Mock;
const mockedStartQueryExecutionCommand = startQueryExecutionCommand as jest.Mock;
const mockedGetQueryResults = getQueryResults as jest.Mock;

const expectedResults = [
    {id:1 , name:test}
]

describe('waitAndGetQueryResults', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });
    it('should retry a failed query and return results when succeeded', async () => {
        const queryId = '123';
        const query = {
            databaseName: 'test',
            queryString: "Select * from test"
        };
        mockedGetQueryExecutionStatus.mockResolvedValueOnce({ state: 'FAILED', stateChangeReason: 'NoSuchKey' } as any)
        mockedStartQueryExecutionCommand.mockResolvedValueOnce('newQueryId');
        mockedGetQueryExecutionStatus.mockResolvedValueOnce({ state: 'SUCCEEDED'} as any)
        mockedGetQueryResults.mockResolvedValue(expectedResults);
        

        const results = await waitAndGetQueryResults(queryId, query);
        expect(mockedGetQueryExecutionStatus).toHaveBeenCalledWith(queryId);
        expect(mockedGetQueryExecutionStatus).toHaveBeenCalledTimes(2);
        expect(mockedStartQueryExecutionCommand).toHaveBeenCalledTimes(1);
        expect(results).toEqual(expectedResults)

    });
});
```

Reply:
Thanks added the tests